server/serverConfigs.go

```diff
@@ -0,0 +1,129 @@
+package server
+
+import (
+	"fmt"
+	toml "github.com/pelletier/go-toml"
+	"github.com/spf13/viper"
+	"log"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+)
+
+var CONFIG_DIR_NAME = ".kelp"
+var cachedConfigDir string
+
+func getProjectFromCmd(cmd []string) string {
+	found := false
+	configsDir := configDirectory()
+
+	for i := range cmd {
+		item := cmd[i]
+
+		if found == true {
+			if strings.HasPrefix(item, configsDir) {
+
+				// strip off prefix, get next path segment
+				item = strings.TrimPrefix(item, configsDir)
+				item = strings.TrimPrefix(item, "/")
+
+				// trim off after /
+				split := strings.SplitN(item, "/", 2)
+
+				if len(split) > 0 {
+					return split[0]
+				}
+
+				return item
+			}
+		} else if item == "--botConf" {
+			found = true
+		}
+	}
+
+	return ""
+}
+
+func configDirectory() string {
+	if len(cachedConfigDir) == 0 {
```

Comment:
I've always heard that it's best to have one return from a function.  Otherwise it becomes really painful to debug if returns are happening in different spots within the function.  I've always done it like this and it's saved lots of debugging time I'm sure, but I can do it that way if you  want.

Reply:
yes please, thanks!